cmd/fossa/cmd/test/test_test.go

```diff
@@ -0,0 +1,86 @@
+package test_test
+
+import (
+	"encoding/json"
+	"flag"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+	"time"
+
+	"github.com/apex/log"
+	"github.com/stretchr/testify/assert"
+	"github.com/urfave/cli"
+
+	"github.com/fossas/fossa-cli/api/fossa"
+	"github.com/fossas/fossa-cli/cmd/fossa/cmd/test"
+)
+
+// taskStatus is a struct that imitates the anonymous struct within fossa.Build
+type taskStatus struct {
+	Status string
+}
+
+type mockServer struct {
+	Responses []interface{}
+}
+
+func (server *mockServer) nextResponse() interface{} {
+	resp := server.Responses[0]
```

Comment:
This is going to panic if the client tries to send more requests than we have responses. Let's make this failure mode more explicit somehow

Reply:
also: should we care about what's in the requests?

Reply:
I started down the path of testing what is in the requests and found that I was mostly testing `url.PathEscape(locator.String())` which requires me to set a lot more information. I do see validity in making another field to store requests, I just didn't think it was necessary here.